src/applications/vaos/utils/data.js

```diff
@@ -183,20 +184,17 @@ export function transformFormToAppointment(state) {
   const data = getFormData(state);
   const clinic = getChosenClinicInfo(state);
   const facility = getChosenFacilityInfo(state);
-  const slot = data.calendarData.selectedDates[0];
+  const slot = getChosenSlot(state);
```

Comment:
We have to get the actual slot to get the end time. We might avoid this in the future by having the selected dates format match the slot format.